File: LC162FindPeakElement.java
package leetcode;

/**
 * https://leetcode.com/problems/find-peak-element/
 */
public class LC162FindPeakElement {
    public int findPeakElement(int[] nums) {
        boolean anteRising = true;
        for (int i = 1; i <= nums.length; i++) {
            boolean postFalling;
            if (i == nums.length) {
                postFalling = true;
            } else {
                postFalling = nums[i - 1] > nums[i];
            }
            if (anteRising && postFalling) {
                return i - 1;
            }
            anteRising = !postFalling;
        }
        return -1;
    }

    public static void main(String[] arguments) {
        LC162FindPeakElement solution = new LC162FindPeakElement();
        System.out.println(solution.findPeakElement(new int[]{1, 2, 3, 1}));
        System.out.println(solution.findPeakElement(new int[]{1, 2, 3, 4}));
    }
}

File: algorithms/code/leetcode/lc162_find_peak_element/lc162_find_peak_element.py
# https://leetcode.com/problems/find-peak-element/

import unittest


class Solution:
    def findPeakElement(self, nums: list[int]) -> int:
        for i in range(1, len(nums) + 1):
            postFalling = None
            postFalling = i == len(nums) or nums[i - 1] > nums[i]
            if postFalling:
                return i - 1
        return -1


class TestCode(unittest.TestCase):
    def test_1_2_3_1(self) -> None:
        assert Solution().findPeakElement([1, 2, 3, 1]) == 2

    def test_1_2_3_4(self) -> None:
        assert Solution().findPeakElement([1, 2, 3, 4]) == 3

    def test_nothing(self) -> None:
        assert Solution().findPeakElement([]) == -1
